daml/ContingentClaims/Claim.daml

```diff
@@ -199,6 +200,23 @@ instance Traversable (ClaimF t x b) where
   sequence (AnytimeF p fa) = AnytimeF p <$> fa
   sequence (UntilF p fa) = UntilF p <$> fa
 
+instance (Show t, Show x, Show a) => Show (Claim t x a) where
+  show = cata \case
+    ZeroF -> "Zero"
+    OneF a -> unwords ["One", show a]
+    GiveF s -> unwords ["Give", bracket s]
+    AndF s s' cs -> print3 "And" s s' cs
+    OrF s s' cs -> print3 "Or" s s' cs
+    CondF b s s' -> unwords ["Cond", bracket (show b), bracket s, bracket  s']
+    ScaleF x s -> print2 "Scale" x s
+    WhenF b s -> print2 "When" b s
+    AnytimeF b s -> print2 "Anytime" b s
+    UntilF b s -> print2 "Until" b s
+    where print3 c s s' cs = unwords [c, bracket s, bracket s', "[" <> intercalate ", " cs <> "]"]
```

Comment:
Yes, what you said is true. And it's trivial to change that. I think even more representative would be to unroll the `And a b [c, d]` into `And a (And b (And c d)))` to match what is in the academic paper.

The problem with all of these approaches is that it makes debugging harder. The `Show` instance is used in `daml test` and the LSP server (although apparently not the scenario output as you pointed out), so any change here would make debugging even more difficult. That's what I meant in the description above:

 > No attempt is made to prettify the output (i.e. changing to infix notation, on changing the constructor names).

I think that the solution to this is to have a separate function, say `showUnrolled` that would provide this functionality.

However, once you begin improving the output, you get closer and closer to the output you get from rendering pricing equations, especially when rendering an `Observation`. e.g. It would make sense to replace `Add` with a `+` in infix form, etc.. And then you eventually end up with an equation. We already have functions for printing that out.